app/models/topic_tag.rb

```diff
@@ -1,22 +1,28 @@
 class TopicTag < ActiveRecord::Base
   belongs_to :topic
-  belongs_to :tag, counter_cache: "topic_count"
+  belongs_to :tag
 
   after_create do
-    if topic&.category_id
-      if stat = CategoryTagStat.where(tag_id: tag_id, category_id: topic.category_id).first
-        stat.increment!(:topic_count)
-      else
-        CategoryTagStat.create(tag_id: tag_id, category_id: topic.category_id, topic_count: 1)
+    if topic && topic.archetype != Archetype.private_message
```

Comment:
Perhaps a test case for this to prevent us from incorrectly increasing the count.